common/metrics/events_reporter.go

```diff
@@ -24,20 +24,32 @@
 
 package metrics
 
-import (
-	"testing"
-
-	"github.com/stretchr/testify/suite"
-)
+import "go.temporal.io/server/common/log"
 
 type (
-	TallyClientTestsSuite struct {
-		MetricTestSuiteBase
+	eventsReporter struct {
+		provider MetricProvider
+		handler  MetricHandler
 	}
 )
 
-func TestTallyClientTest(t *testing.T) {
-	s := new(TallyClientTestsSuite)
-	s.MetricTestUtilityProvider = func() MetricTestUtility { return NewTallyMetricTestUtility() }
-	suite.Run(t, s)
+var _ Reporter = (*eventsReporter)(nil)
+
+func NewEventsReporter(h MetricHandler) eventsReporter {
```

Comment:
I assume this will work with custom reporter extension point? Please update comments on WithCustomMetricsReporter()